mm2src/mm2_main/src/lp_swap/taker_swap.rs

```diff
@@ -1404,24 +1408,29 @@ impl TakerSwap {
             && self.taker_coin.is_supported_by_watchers()
             && self.maker_coin.is_supported_by_watchers()
         {
-            let taker_spends_maker_payment_fut = self.taker_coin.create_taker_spends_maker_payment_preimage(
+            let maker_payment_spend_preimage_fut = self.taker_coin.create_maker_payment_spend_preimage(
                 &self.r().maker_payment.as_ref().unwrap().tx_hex,
                 self.maker_payment_lock.load(Ordering::Relaxed) as u32,
                 self.r().other_maker_coin_htlc_pub.as_slice(),
                 &self.r().secret_hash.0,
                 &self.unique_swap_data()[..],
             );
-            let taker_refunds_payment_fut = self.taker_coin.create_taker_refunds_payment_preimage(
+
+            let time_lock = match std::env::var("USE_TEST_LOCKTIME") {
+                Ok(_) => self.r().data.started_at as u32,
+                Err(_) => self.r().data.taker_payment_lock as u32,
```

Comment:
We should use `into_or` or `into_or_max` to handle overflow here.

Reply:
Since these are timestamps, I think it's even better to manually use `try_from` and return error if can't cast the type.